include/room_member.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 Tunnler Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+//
+
+#pragma once
+
+#include <deque>
+#include <string>
+
+#include "tunnler.h"
+
+#include "RakNetStatistics.h"
+#include "RakPeerInterface.h"
+
+class RoomMember final {
+public:
+
+	struct MemberInformation {
+		std::string nickname;
+		std::string game_name;
+		MacAddress mac_address;
+	};
+
+	enum class State {
+
+		//Actual states
+		StateIdle, // Default state
+		StateError, // Some error [permissions to network device missing or something]
+		StateJoining, // This is also while the server didn't receive a setup packet / nickname from this player
+		StateJoined,
+    
+		// Reasons for connection loss
+		StateLostConnection,
+		StateKicked,
+    
+		// Reasons why connection was rejected
+		StateRoomFull,
+		StateRoomDestroyed,
+		StateWrongVersion,
+		StateWrongPassword,
+		StateBanned,
+		StatePasswordRequired,
+		StateNicknameAlreadyInUse,
+	};
+
+	struct ChatEntry {
+		std::string nickname;
+		std::string message;
+	};
+
+	RoomMember(MemberInformation member_information);
```

Comment:
What do we need these constructors for? I was under the impression that the RoomMember object should always be constructed without parameters, and only be filled once you call Join()